core/chains/evm/log/registrations_unit_test.go

```diff
@@ -54,3 +72,100 @@ func TestRegistrationsUnit_PanicsOnDoubleSubscribe(t *testing.T) {
 	// Now we can add it again
 	r.addSubscriber(sub)
 }
+
+func newTopic() Topic {
+	return Topic(utils.NewHash())
+}
+
+func TestRegistrationsUnit_addSubscriber(t *testing.T) {
+	contractAddr := testutils.NewAddress()
+	r := newTestRegistrations(t)
+
+	l := newTestListener(t, 1)
+	topic1 := utils.NewHash()
+	topicValueFilters1 := [][]Topic{[]Topic{newTopic(), newTopic()}, []Topic{newTopic()}, []Topic{}}
+	topic2 := utils.NewHash()
+	topicValueFilters2 := [][]Topic{[]Topic{newTopic()}}
+	topic3 := utils.NewHash()
+	topicValueFilters3 := [][]Topic{}
+	logsWithTopics := make(map[common.Hash][][]Topic)
+	logsWithTopics[topic1] = topicValueFilters1
+	logsWithTopics[topic2] = topicValueFilters2
+	logsWithTopics[topic3] = topicValueFilters3
+	opts := ListenerOpts{Contract: contractAddr, LogsWithTopics: logsWithTopics, MinIncomingConfirmations: 1}
+	sub := &subscriber{l, opts}
+	r.addSubscriber(sub)
+
+	// same contract, same topics
+	l2 := newTestListener(t, 2)
+	opts2 := opts
+	sub2 := &subscriber{l2, opts2}
+	r.addSubscriber(sub2)
+
+	// same contract, different topics
+	l3 := newTestListener(t, 3)
+	topic4 := utils.NewHash()
+	topicValueFilters4 := [][]Topic{[]Topic{newTopic()}}
+	logsWithTopics3 := make(map[common.Hash][][]Topic)
+	logsWithTopics3[topic4] = topicValueFilters4
+	opts3 := opts
+	opts3.LogsWithTopics = logsWithTopics3
+	sub3 := &subscriber{l3, opts3}
+	r.addSubscriber(sub3)
+
+	assert.Equal(t, 1, int(r.highestNumConfirmations))
+
+	// same contract, same topics, different MinIncomingConfirmations
+	l4 := newTestListener(t, 4)
+	opts4 := opts3
+	opts4.MinIncomingConfirmations = 42
+	sub4 := &subscriber{l4, opts4}
+	r.addSubscriber(sub4)
+
+	assert.Equal(t, 42, int(r.highestNumConfirmations))
+
+	assert.Len(t, r.registeredSubs, 4)
```

Comment:
You could replace this whole block of 5 lines with something like this:

require.ElementsMatch(t, [4]{sub, sub2, sub3, sub4}, registeredSubs)

Reply:
And same in below blocks too

Reply:
ElementsMatch doesn't work because this is a map not a slice/array